### PR TITLE
Preparation for kubernetes branch merging

### PR DESCRIPTION
**Pull request by phantomjinx**
Adds a clienttools object to encapsulate all clients initialised in the operator.

### CODE REVIEW

File: install/operator/pkg/syndesis/configuration/configuration.go [hunk 1]
```diff
@@ -297,6 +306,45 @@ func GetAddons(configuration Config) []AddonInstance {
 	}
 }
 
+/*
+ * For testing if the giving platform is Openshift
+ */
+func ApiCapabilities(apiClient kubernetes.Interface) (*ApiServerSpec, error) {
```
Comment (lgarciaaco) [inline install/operator/pkg/syndesis/configuration/configuration.go:312]: I would argue that this is the right place for this function

File: install/operator/.lib.sh [hunk 2]
```diff
@@ -116,13 +116,22 @@ build_operator()
...
-          for GOOS in linux darwin windows ; do
+          for GOOS in ${GOOSLIST} ; do
             export GOARCH GOOS
             echo building ./dist/${GOOS}-${GOARCH}/syndesis-operator executable
             go build  "$@" -o ./dist/${GOOS}-${GOARCH}/syndesis-operator \
                 -gcflags all=-trimpath=${GOPATH} -asmflags all=-trimpath=${GOPATH} -mod=vendor \
                 ./cmd/manager
+
+            echo building ./dist/${GOOS}-${GOARCH}/platform-detect executable
+            go build -o ./dist/${GOOS}-${GOARCH}/platform-detect \
```
Comment (lgarciaaco) [inline install/operator/.lib.sh:132]: What is the use of this binary?

Comment (phantomjinx) [reply install/operator/.lib.sh:132]: platform-detect externalises the ApiCapabilities function for use in the syndesis cli scripts. You haven't seen those changes yet (haven't finalised them).

File: install/operator/pkg/syndesis/configuration/configuration.go (hunk 1 above)
Comment (phantomjinx) [reply install/operator/pkg/syndesis/configuration/configuration.go:312]: ok?

**Comment by phantomjinx** [comment]
> @phantomjinx Did you test backup and upgrade scenarios when you include the changes?

Tested backup but not upgrade.

**Comment by phantomjinx** [comment]
> > @phantomjinx Did you test backup and upgrade scenarios when you include the changes?
> 
> Tested backup but not upgrade.

Tested both now. Both ok.

File: install/operator/pkg/syndesis/configuration/configuration.go (hunk 1 above)
Comment (phantomjinx) [reply install/operator/pkg/syndesis/configuration/configuration.go]: Moved ApiCapabilities to its own package.

**Pull request merged**